Add coverage for Puppet Module content search functionality

oshtaier opened the pull request:
Closes #3496 

Adding basic coverage that can be expanded in future (however not sure what scenarios can be added)

```
nosetests tests/foreman/ui/test_puppetmodule.py
...
----------------------------------------------------------------------
Ran 3 tests in 100.951s

OK
```

abalakh commented on robottelo/ui/locators/base.py:
```diff
@@ -1779,6 +1779,38 @@
+    # Puppet Module
+    "puppet.module_name": (By.XPATH, "//a[contains(., '%s')]"),
+    "puppet.author": (
+        By.XPATH, "//span[text()='Author']/../"
```
I bet you tried, but... Isn't there a way not to stick to text values (which will depend on locale settings, of course)?

oshtaier replied:
Nope =))

abalakh commented on robottelo/ui/puppetmodule.py:
```diff
@@ -0,0 +1,60 @@
+        """Check whether puppet module is present in necessary
+        repository/repositories
+        """
+        self.click(self.search(name))
```
nitpick: `self.search_and_click()`?

oshtaier replied:
hate that helper, but okay =)

abalakh commented on robottelo/ui/puppetmodule.py:
```diff
@@ -0,0 +1,60 @@
+            searchbox = self.wait_until_element(locators['repo.search'])
+            if searchbox:
+                searchbox.clear()
+                searchbox.send_keys(repo)
```
Why not `self.assign_value(searchbox, repo)` here?

oshtaier replied:
dumb copy paste =)

abalakh commented on tests/foreman/ui/test_puppetmodule.py:
```diff
@@ -0,0 +1,104 @@
+        cls.organization = entities.Organization().create()
+        product = entities.Product(organization=cls.organization).create()
+        cls.repo = entities.Repository(
+            name=gen_string('alpha'),
```
Any specific reason for this line?

oshtaier replied:
doubt that any reason for it

abalakh commented on tests/foreman/ui/test_puppetmodule.py:
```diff
@@ -0,0 +1,104 @@
+    @classmethod
+    def setUpClass(cls):
+        super(PuppetModuleTestCase, cls).setUpClass()
+        cls.organization = entities.Organization().create()
```
Please use session org instead

abalakh commented on tests/foreman/ui/test_puppetmodule.py:
```diff
@@ -0,0 +1,104 @@
+        puppet modules are present inside of repository
+        """
+        with Session(self.browser) as session:
+            session.nav.go_to_select_org(self.organization.name)
```
This line should be dropped when session_org approach is applied

abalakh commented on tests/foreman/ui/test_puppetmodule.py:
```diff
@@ -0,0 +1,104 @@
+                'ntp',
+                [
+                    ['Author', 'puppetlabs'],
+                    ['Version', '3.2.1'],
```
Hm... TBH i don't like the idea to hardcode such values in tests. It's not the first and only one test with such values and in case we use different archive (which can be the case, as we predefined `PUPPET_MODULE_NTP_PUPPETLABS` constant so its value can be changed later) we'll have to fix the tests. Maybe it's better to create a dict in constants.py with all these values so in case we change the constant it will be clear that other values should be updated too? I know the idea is so-so, feel free to suggest something better.

oshtaier replied:
No changes to module since start of the project, so I suggest to leave like it is. We can avoid any removal or upgrade of the module from our own code folder.
Anyway, I will think about that. Thanks

oshtaier replied:
Decided to leave like it is as this file present locally, not even in fake repository

oshtaier commented:
Addressed comments:
```
nosetests tests/foreman/ui/test_puppetmodule.py
...
----------------------------------------------------------------------
Ran 3 tests in 98.046s

OK
```

sghai commented:
ACK
